packages/core/src/spam-protection/spam-protection-action-creator.ts

```diff
@@ -99,32 +99,53 @@ export default class SpamProtectionActionCreator {
     }
 
     execute(): ThunkAction<SpamProtectionAction, InternalCheckoutSelectors> {
-        return (store) =>
-            concat(
-                of(createAction(SpamProtectionActionType.ExecuteRequested)),
-                this.initialize()(store),
-                this._googleRecaptcha
-                    .execute()
-                    .pipe(take(1))
-                    .pipe(
-                        switchMap(async ({ error, token }) => {
-                            if (error instanceof SpamProtectionChallengeNotCompletedError) {
-                                throw error;
-                            }
+        return (store) => {
+            const { config, checkout } = store.getState();
+            const {
+                checkoutSettings: { features },
+            } = config.getStoreConfigOrThrow();
+            const { shouldExecuteSpamCheck } = checkout.getCheckoutOrThrow();
+
+            const recaptchaExperiment = features['CHECKOUT-8264.recaptcha_error_fix'];
+
+            if (recaptchaExperiment) {
+                if (shouldExecuteSpamCheck) {
+                    return this._executeRecaptcha(store);
+                }
+            } else {
+                return of(createAction(SpamProtectionActionType.ExecuteActionNotNeeded, undefined));
```

Comment:
Could we change the `CustomerActionCreator` to check the `shouldExecuteSpamCheck` flag instead, similar to how the check is implemented in `PaymentStrategyActionCreator`: https://github.com/bigcommerce/checkout-sdk-js/blob/517152be10a54238079315e99dc67ac5bb438b4a/packages/core/src/payment/payment-strategy-action-creator.ts#L60-L62

This approach will reduce the risk of change as it won't impact the payment step, which already has a check in place as shown above.

Reply:
Thanks DC. That seems good, that was my first thought as well 😄